packages/block-library/src/utils/hooks.js

```diff
@@ -19,6 +22,53 @@ export function useCanEditEntity( kind, name, recordId ) {
 	);
 }
 
-export default {
-	useCanEditEntity,
-};
+/**
+ * Handles uploading a media file from a blob URL on mount.
+ *
+ * @param {Object}   args              Upload media arguments.
+ * @param {string}   args.url          Blob URL.
+ * @param {?Array}   args.allowedTypes Array of allowed media types.
+ * @param {Function} args.onChange     Function called when the media is uploaded.
+ * @param {Function} args.onError      Function called when an error happens.
+ */
+export function useUploadMediaFromBlobURL( args = {} ) {
+	const latestArgs = useRef( args );
+	const { getSettings } = useSelect( blockEditorStore );
+
+	useLayoutEffect( () => {
+		latestArgs.current = args;
+	} );
```

Comment:
This implementation is to avoid unnecessary side effects and ensure that the process is only executed when the block is mounted, right?

Reply:
This is common pattern. See - https://epicreact.dev/the-latest-ref-pattern-in-react/